### DISCUSSION TITLE
Does deleting an Album delete the photos

### DISCUSSION THREAD

**Discussion by bcrust** [immich-app/immich#12609] [Q&A]
I accidentally uploaded my daughter photos with my ID and put them in an album.
To her dismay, I can see them.  I downloaded them and re-uploaded them as her user to a new album and would like to delete from from my account.

I figured the easiest way would be to delete the album, but I am still able to see the pictures in the /Photo tab.  If I delete them from /Photos, she is able to see her copies in her album under her account.

So, was I wrong to assume that deleting the Album also deletes the Photos?

Is there a mechanism to delete the Photos easily, despite the fact that I already deleted the album?

I am running the latest chart (4.0.15 = App Version 1.114.0 on Truenas).

Thanks in advance.

**Comment by bo0tzz** [comment]
> So, was I wrong to assume that deleting the Album also deletes the Photos?

Yes, an album is just a collection. If you want to delete all the photos in one, you can select-all and delete them before removing the album.

**Reply by bcrust** [reply]
So, I blew it twice.  Once by using the wrong account and the second time by deleting the album first.

OK.  I can delete the asset from assets where the ownerId is me and the createdAt is yesterday.
Is that sufficient or do other tables need to be cleaned up, too?

**Comment by frknkrc44** [comment]
You can use this Python code to delete all not-in-album assets

```python
from json import dumps
from requests import post, delete

SERVER_URL = '<SERVER-URL-HERE>'
API_KEY = '<API-KEY-HERE>'
API_PREFIX = '/api'
SEARCH_PHOTOS_URL = f'{SERVER_URL}{API_PREFIX}/search/metadata'
ASSETS_URL = f'{SERVER_URL}{API_PREFIX}/assets'
API_HEADERS = {
    'Content-Type': 'application/json',
    'Accept': 'application/json',
}

while True:
    req = post(
        f'{SEARCH_PHOTOS_URL}?apiKey={API_KEY}',
        headers=API_HEADERS,
        data=dumps({ 'isNotInAlbum': True }),
    )

    asset_ids = [item['id'] for item in req.json()['assets']['items']]
    if not len(asset_ids):
        print('All photos in albums')
        break

    result = delete(
        f'{ASSETS_URL}?apiKey={API_KEY}',
        headers=API_HEADERS,
        data=dumps({ 'ids': asset_ids }),
    )

    print('Delete result:', result.json())
```
